src/main/java/bio/terra/service/snapshot/flight/create/CreateSamGroupStep.java

```diff
@@ -0,0 +1,30 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+
+public class CreateSamGroupStep implements Step {
+  private final IamService iamClient;
+  private final SnapshotRequestModel snapshotReq;
+
+  public CreateSamGroupStep(IamService iamClient, SnapshotRequestModel snapshotReq) {
+    this.iamClient = iamClient;
+    this.snapshotReq = snapshotReq;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    iamClient.createGroup(snapshotReq.getName());
```

Comment:
Sam group names must be globally unique.  So if another group exists with this name, this call will fail.  What should the behavior be in that case?

Most critically, If TDR is an administrator of that same-named group, then `undoStep` below will errantly delete it.

You might like to take a look at how Firecloud group creation is handled as part of the DUOS-TDR integration: https://github.com/DataBiosphere/jade-data-repo/blob/75bc900f725b15fc3b573bd996ff88361f8d314e/src/main/java/bio/terra/service/snapshot/flight/duos/CreateDuosFirecloudGroupStep.java

Reply:
This is a good point! I am thinking since request names must be unique, and for byRequestId mode, they will contain the snapshot access request uuid, there shouldn't be a sam group existing with that name. But, you bring up a good point because it is possible. I'll bring this up with my team! I'll also definitely follow the approach used there, that looks like a great idea.